py3status/modules/group.py

```diff
@@ -220,11 +222,17 @@ def group(self):
             current_output = self._get_output()
             # if the output is empty try and find some output
             if not current_output:
+                if self._first_run:
+                    self._first_run = False
+                    return {
+                        'cached_until': self.py3.time_in(MAX_NO_CONTENT_WAIT),
```

Comment:
IIRC we should avoid prefix `_` to keep the modules clean... and we should carry things all the way to the return rather than using multiple returns... eg `current_output = [{'full_text': ''}]` for more free/flexible customization as we have `{button}` placeholder too.

Reply:
> IIRC we should avoid prefix _ to keep the modules clean.

in my mind `_first_run` is private so it is marked as such.

> rather than using multiple returns

I think it depends.  Here it would have made the PR much more complicated so I'm ok with it but let's see what @ultrabug has to say

Reply:
See if you like this diff.
```
diff --git a/py3status/modules/group.py b/py3status/modules/group.py
index bebae3dd..07a5bd9e 100644
--- a/py3status/modules/group.py
+++ b/py3status/modules/group.py
@@ -118,6 +118,7 @@ class Py3status:
         if not self.items:
             self.cycle = 0
 
+        self.first_run = True
         self.active = 0
         self.last_active = 0
         self.urgent = False
@@ -220,11 +221,14 @@ class Py3status:
             current_output = self._get_output()
             # if the output is empty try and find some output
             if not current_output:
-                self._change_active(1)
-                current_output = self._get_output()
-                # there is no output for any module retry later
-                if not current_output:
-                    update_time = RETRY_TIMEOUT_NO_CONTENT
+                if self.first_run:
+                    update_time = 1
+                else:
+                    self._change_active(1)
+                    current_output = self._get_output()
+                    # there is no output for any module retry later
+                    if not current_output:
+                        update_time = RETRY_TIMEOUT_NO_CONTENT
         else:
             urgent = self.urgent
             current_output = []
@@ -237,6 +241,7 @@ class Py3status:
             format_control = self.format_button_closed
             format = self.format_closed
 
+        self.first_run = False
         button = {'full_text': format_control, 'index': 'button'}
         composites = {
             'output': self.py3.composite_create(current_output),
```